components/resourceDetails/resourceSharing/agentAccessTable/index.jsx

```diff
@@ -116,47 +115,28 @@ export default function AgentAccessTable({ type, loading, setLoading }) {
   // };
 
   if (!tablePermissions.length && isCustomPolicy(type)) return null;
-
   const { emptyStateText } = POLICIES_TYPE_MAP[type];
 
-  return (
-    <Accordion
-      defaultExpanded
-      classes={{ root: classes.accordion, rounded: classes.rounded }}
-      data-testid={TESTCAFE_ID_AGENT_ACCESS_TABLE}
-    >
-      <PolicyHeader type={type} isPolicyList>
-        <>
-          <AddAgentButton
-            type={type}
-            setLoading={setLoading}
-            permissions={tablePermissions}
-          />
-          <PolicyActionButton
-            permissions={tablePermissions}
-            setLoading={setLoading}
-            type={type}
-          />
-        </>
-      </PolicyHeader>
-      <div className={classes.permissionsContainer}>
-        {!!tablePermissions.length && (
-          <>
-            {/* TODO: Uncomment to reintroduce tabs */}
-            {/* <Tabs */}
-            {/*  handleTabChange={handleTabChange} */}
-            {/*  selectedTabValue={selectedTabValue} */}
-            {/*  tabs={[{ label: "All", value: "all", testid: TESTCAFE_ID_TAB_ALL },  { label: "People", value: "agent", testid: TESTCAFE_ID_TAB_PEOPLE }, { label: "Groups", value: "group", testid: TESTCAFE_ID_TAB_GROUPS } }]} */}
-            {/* /> */}
-            <AgentsSearchBar handleFilterChange={handleFilterChange} />
-          </>
-        )}
-        {loading && (
-          <Container variant="empty">
-            <CircularProgress />
-          </Container>
-        )}
-        {!loading && tablePermissions.length ? (
+  const tableContents = () => {
```

Comment:
would be better to make this an explicit component, but keep it in the same file, that way you've props you can pass to it, and it won't be redeclared each render.

Reply:
Awesome - makes sense. Thanks!